### PR TITLE
ansible-lint: add package formula

### PR DESCRIPTION
**Pull request by Nklya**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
This PR adds ansible-lint package formula.
ansible-lint checks playbooks for practices and behaviour that could potentially be improved.
Repository: https://github.com/willthames/ansible-lint

-----
Previously, i tried to add ansible-lint into ansible package, but it was rejected, so i made separate package. https://github.com/Homebrew/homebrew-core/pull/12197

### CODE REVIEW

**Comment by tgagor** [comment]
+1 for this work

File: Formula/ansible-lint.rb [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    url "https://files.pythonhosted.org/packages/b3/b2/238e2590826bfdd113244a40d9d3eb26918bd798fc187e2360a8367068db/six-1.10.0.tar.gz"
+    sha256 "105f8d68616f8248e24bf0e9372ef04d3cc10104f1980f54d57b2ce73a5ad56a"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system "#{bin}/ansible-lint", "--version"
```
Comment (bfontaine) [inline Formula/ansible-lint.rb]: Please write a test that does something more substantial than e.g. `--version` or `--help`. You could write a simple playbook and check it for errors. Please ask if you need help.

**Review by bfontaine — CHANGES_REQUESTED** [review]

File: Formula/ansible-lint.rb [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    ENV["ANSIBLE_REMOTE_TEMP"] = testpath/"tmp"
+    (testpath/"playbook.yml").write <<-EOF.undent
+      ---
+      - hosts: all
+        gather_facts: False
+        tasks:
+        - name: ping
+          ping:
+    EOF
+    system bin/"ansible-lint", '-v',testpath/"playbook.yml"
```
Comment (bfontaine) [inline Formula/ansible-lint.rb]: Please use double-quotes everywhere: `system bin/"ansible-lint", "-v", testpath/"playbook.yml"`

**Comment by Nklya** [comment]
I've add tests and fix quotes. 
Actually, "-v" option is not needed.

**Review by bfontaine — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @Nklya for contributing this new formula to Homebrew! We look forward to seeing other pull requests from you :)

**Comment by Nklya** [comment]
Thanks @bfontaine and @fxcoudert for reviewing and approval.

**Pull request closed without merging**